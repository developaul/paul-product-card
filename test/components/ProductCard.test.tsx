import React from 'react'
import renderer, { act } from 'react-test-renderer'

import ProductCard from '../../src/components';

import { product2 } from './data/products';

describe('ProductTitle', () => {
  test('It should display the component correctly', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {() => (
          <h1>Product Card</h1>
        )}
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  test('It should increment the counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}></button>
          </>
        )}
      </ProductCard>
    )

    let tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    })

    tree = wrapper.toJSON()

    expect((tree as any).children[1].children[0]).toBe('1')
  })
})