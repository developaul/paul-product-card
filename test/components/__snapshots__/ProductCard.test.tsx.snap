// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle It should display the component correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;

exports[`ProductTitle It should increment the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
